pages/index.js

```diff
@@ -1,27 +1,109 @@
-import Card, {
-  CardHeader,
-  CardTitle,
-  CardContent,
-  CardButtonContainer,
-} from '../components/card';
-import { EDIT } from '../constants';
-import Button from '../components/button';
+import firebase from 'firebase'
+import styled from 'styled-components'
+import React, { useState } from 'react'
+import { useRouter } from 'next/router'
+import { COLOR } from '../constants'
+import { checkAdminClaim } from '../utility/auth'
+import background from '../public/backgroundImage.png'
+import nwPlusLogo from '../public/nwPlusLogo.png'
+import signIn from '../public/signIn.png'
+
+const BackgroundContainer = styled.div`
+    background-image: url(${background});
+    background-size: cover;
+    height: 100vh;
+    width: 100vw;
+`
+
+const LeftContainer = styled.div`
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    height: 100vh;
+    width: 337px;
+    background-color: white;
+`
+
+const LogoImage = styled.img`
+    align-self: center;
+    margin-top: 80px;
+`
+
+const LogInDiv = styled.div`
+    font-size: 28px;
+    margin-top: 48px;
+`
+
+const SignInImage = styled.img`
+    align-self: center;
+    margin-top: 34px;
+    cursor: pointer;
+`
+
+const StatusDiv = styled.div`
+    margin-top: 20px;
+    font-size: 26px;
+    text-align: center;
+    ${(props) => props.error && `color: ${COLOR.RED}`}
+`
 
 export default function Home() {
+  const router = useRouter()
+  const [ showError, setShowError ] = useState(false)
+  const [ isAddingClaim, setIsAddingClaim ] = useState(false)
+  const setAdmin = firebase.functions().httpsCallable('setAdmin')
+
+  const googleSignIn = async () => {
+    const provider = new firebase.auth.GoogleAuthProvider();
+    provider.setCustomParameters({
+      'hd': 'nwplus.io',
+      'prompt': 'consent'
+    })
+
+    try {
+      await firebase.auth().signInWithPopup(provider)
+      setShowError(false)
+      setIsAddingClaim(true)
+      const user = await firebase.auth().currentUser
+      const isAdmin = await checkAdminClaim(user)
+      if (isAdmin) {
+        await router.push('/landing')
+      } else {
+        const res = await setAdmin()
+        if (res.data.isAdmin) {
+          await user.getIdToken(true)
+          await router.push('/landing')
+        } else {
+          await firebase.auth().signOut()
+          await router.push('/')
```

Comment:
I think `await` before `router.push` is not needed
```suggestion
        router.push('/landing')
      } else {
        const res = await setAdmin()
        if (res.data.isAdmin) {
          await user.getIdToken(true)
          router.push('/landing')
        } else {
          await firebase.auth().signOut()
          router.push('/')
```

Reply:
You're right that I don't need the `await` in the login page, but I think I should still use it in auth.js. It seems like `router.push` is an asynchronous operation, so if a user that is not logged in tries to access the landing page, `setIsLoading(false)` might run first before `router.push` finishes pushing that user back to the login page, which results in a brief flash of protected content.

Reply:
Oh ok makes sense, gotcha 👍